src/core/quantum_gate.jl

```diff
@@ -174,13 +117,112 @@ function apply_gate!(state::Ket, gate::AbstractGate)
     operator=get_operator(gate,type_in_ket)
 
     apply_operator!(state,operator,connected_qubits)
+end
+
+# specialization for single target dense gate (size N=2, for N=2^target_count)
+function apply_operator!(
```

Comment:
Are you using any specific reference for this implementation? If so, please cite it here.